src/sortTemplate.ts

```diff
@@ -50,81 +50,89 @@ export async function sortTemplate(template: DeploymentTemplate | undefined, sec
     ext.outputChannel.appendLine("Sorting template");
     switch (sectionType) {
         case TemplateSectionType.Functions:
-            await sortFunctions(template, textEditor);
+            await showSortingResultMessage(() => sortFunctions(template, textEditor), "Functions");
             break;
         case TemplateSectionType.Outputs:
-            await sortOutputs(template, textEditor);
+            await showSortingResultMessage(() => sortOutputs(template, textEditor), "Outputs");
             break;
         case TemplateSectionType.Parameters:
-            await sortParameters(template, textEditor);
+            await showSortingResultMessage(() => sortParameters(template, textEditor), "Parameters");
             break;
         case TemplateSectionType.Resources:
-            await sortResources(template, textEditor);
+            await showSortingResultMessage(() => sortResources(template, textEditor), "Resources");
             break;
         case TemplateSectionType.Variables:
-            await sortVariables(template, textEditor);
+            await showSortingResultMessage(() => sortVariables(template, textEditor), "Variables");
             break;
         case TemplateSectionType.TopLevel:
-            await sortTopLevel(template, textEditor);
+            await showSortingResultMessage(() => sortTopLevel(template, textEditor), "Top level items");
             break;
         default:
             assertNever(sectionType);
-
     }
-    vscode.window.showInformationMessage("Done sorting template!");
 }
 
-async function sortOutputs(template: DeploymentTemplate, textEditor: vscode.TextEditor): Promise<void> {
+async function showSortingResultMessage(sortAction: () => Promise<boolean>, part: string): Promise<void> {
+    let didSorting = await sortAction();
+    let message = didSorting ? `${part} were sorted` : `No ${part.toLowerCase()} needed sorting`;
```

Comment:
nit: Don't know if we were ever do localization, but this is English-specific code so it will cause problems if we do.  Better options would be:
1) Pass in both messages ("Resources were sorted" and "No resources needed sorting")
2) Make it more generic, e.g. pass in "resources" and use something like `"${part}" section was sorted` and `Nothing in "${part}" needed sorting"

Reply:
I tried to implement option 2 but was not really sure what you meant.

Reply:
If "No ${part} needed sorting" were to be translated into German it would be something like "Kein ${part} braucht sortiert zu werden".  Replace ${part} with "resource" and you get "Kein resource braucht sortiert zu werden", which is not German.  You either need to give the localizers full strings such as "No resources needed sorting" and "Resources were sorted" (option 1), or generalize to something like `Nothing in "${part}" needed sorting` (note the double quotes), so you get `Kein Element in "resources" bracht sortiert zu werden`, which is acceptable in German, esp. since they're already working with "resources" as an English keyword.  Make sense?

Reply:
Yes. I have fixed the double quotes in the messages now.